Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift

```diff
@@ -0,0 +1,159 @@
+                serviceName = serviceNamespace + "." + serviceName!
+            }
+            if localEndpointCache[serviceName!] == nil {
+                localEndpoint = defaultLocalEndpoint.clone(serviceName: serviceName!)
+                localEndpointCache[serviceName!] = localEndpoint
+            }
+        }
+
+        var remoteEndpoint: ZipkinEndpoint?
+        if (otelSpan.kind == .client || otelSpan.kind == .producer) && attributeEnumerationState.RemoteEndpointServiceName != nil {
+            remoteEndpoint = remoteEndpointCache[attributeEnumerationState.RemoteEndpointServiceName!]
+            if remoteEndpoint == nil {
+                remoteEndpoint = ZipkinEndpoint(serviceName: attributeEnumerationState.RemoteEndpointServiceName!)
+                remoteEndpointCache[attributeEnumerationState.RemoteEndpointServiceName!] = remoteEndpoint!
+            }
+        }
+
+        let status = otelSpan.status
+
+        if status?.isOk ?? false {
+            attributeEnumerationState.tags[statusCode] = "\(status!.canonicalCode)".capitalized
+            if status?.statusDescription != nil {
+                attributeEnumerationState.tags[statusDescription] = status!.description
+            }
+        }
+
+        let annotations = otelSpan.timedEvents.map { processEvents(event: $0) }
+
+        return ZipkinSpan(traceId: ZipkinConversionExtension.EncodeTraceId(traceId: otelSpan.traceId, useShortTraceIds: useShortTraceIds),
+                          parentId: parentId,
+                          id: ZipkinConversionExtension.EncodeSpanId(spanId: otelSpan.spanId),
+                          kind: ZipkinConversionExtension.toSpanKind(otelSpan: otelSpan),
+                          name: otelSpan.name,
+                          timestamp: otelSpan.startEpochNanos / 1000,
+                          duration: ( otelSpan.endEpochNanos - otelSpan.startEpochNanos) * 1000000,
+                          localEndpoint: localEndpoint,
+                          remoteEndpoint: remoteEndpoint,
+                          annotations: annotations,
+                          tags: attributeEnumerationState.tags,
+                          debug: nil,
+                          shared: nil)
+    }
+
+    static func EncodeSpanId(spanId: SpanId) -> String {
+        return spanId.hexString
+    }
+
+    private static func EncodeTraceId(traceId: TraceId, useShortTraceIds: Bool) -> String {
+        if useShortTraceIds {
+            return String(format: "%016llx", traceId.lowerLong)
+        } else {
+            return traceId.hexString
+        }
+    }
+
+    private static func toSpanKind(otelSpan: SpanData) -> String {
+        switch otelSpan.kind {
+        case .server:
+            return "SERVER"
+        case .producer:
+            return "PRODUCER"
+        case .consumer:
+            return "CONSUMER"
+        default:
+            return "CLIENT"
```

Comment:
I think default is empty string or something. Please refer to the spec

Reply:
True, should not be set